sparklightautoml/ml_algo/base.py

```diff
@@ -266,6 +257,46 @@ def _combine_val_preds(self, val_data: SparkDataset, val_preds: Sequence[SparkDa
 
         return full_val_preds
 
+    def _parallel_fit(self, train_valid_iterator: SparkBaseTrainValidIterator) \
+            -> Tuple[List[Model], List[SparkDataFrame], List[str]]:
+        num_folds = len(train_valid_iterator)
+
+        def build_fit_func(i: int, mdl_pred_col: str, train: SparkDataset):
+            def func() -> Optional[Tuple[int, Model, SparkDataFrame, str]]:
+                if num_folds > 1:
+                    logger.info2(
+                        "===== Start working with \x1b[1mfold {}\x1b[0m for \x1b[1m{}\x1b[0m "
+                        "=====".format(i, self._name)
+                    )
+
+                if self.timer.time_limit_exceeded():
+                    logger.info(f"No time to calculate fold {i}/{num_folds} (Time limit is already exceeded)")
+                    return None
+
+                mdl, vpred, _ = self.fit_predict_single_fold(mdl_pred_col, self.validation_column, train)
+                vpred = vpred.select(SparkDataset.ID_COLUMN, train.target_column, mdl_pred_col)
+
+                return i, mdl, vpred, mdl_pred_col
+            return func
+
+        fit_tasks = [
```

Comment:
here and below there a lot of such pattern - create fit_tasks, and then compute those. please consider making it in more general form, e.g through class inheritance